tsg/ranker/base.py

```diff
@@ -78,19 +119,29 @@ def cosine_score_calc(query_terms, index_dictionary_path=DICTIONARY_PATH,
 
     return scores
 
+def combine_and_or_scores(and_dict, or_dict):
+    sorted_and = sorted(and_dict.items(), key = operator.itemgetter(1,0))
+    sorted_or = sorted(or_dict.items(), key = operator.itemgetter(1, 0))
+
+    new_docs = set(sorted_or) - set(sorted_and)
```

Comment:
You are confused. Here we are getting those documents that belongs to Sorted_or  but don't belong to sorted_and. The idea is to get those exact match (docs in AND) first then we add the new_docs.